### PR TITLE
docs: Metrics tag openapi docs

### PR DESCRIPTION
**Pull request by chriswk**
I've copied most of the descriptions from what we did for batch metrics under the Edge tag, however there are a couple of top level descriptions that are "fine" but I'm definitely open to suggestions.

### CODE REVIEW

File: src/lib/openapi/spec/application-schema.ts [hunk 1]
```diff
@@ -7,31 +7,51 @@ export const applicationSchema = {
...
             type: 'string',
         },
         url: {
+            description:
+                'A link to reference the application reporting the metrics. Could for instance be a GitHub link to the repository of the application',
             type: 'string',
+            example: 'https://github.com/Unleash/unleash-client-proxy-js',
         },
         color: {
+            description: `Css color to be used to color the application's entry in the application list`,
```
Comment (thomasheartman) [inline src/lib/openapi/spec/application-schema.ts:41]:
```suggestion
            description: `The CSS color that is used to color the application's entry in the application list`,
```

File: src/lib/openapi/spec/application-schema.ts
Comment (thomasheartman) [inline src/lib/openapi/spec/application-schema.ts:1]: Is this a request or response schema? That might affect the verbiage a bit. For instance: "the x that should be used" makes more sense in a request schema, while "the x that is used" makes more sense in a response schema. This is nit stuff, though, and absolutely non-blocking.

File: src/lib/openapi/spec/application-schema.ts [hunk 2]
```diff
@@ -7,31 +7,51 @@ export const applicationSchema = {
     required: ['appName'],
```
Comment (thomasheartman) [inline src/lib/openapi/spec/application-schema.ts:7]: Are there other properties that are always present?

File: src/lib/openapi/spec/application-schema.ts [hunk 3]
```diff
@@ -7,31 +7,51 @@ export const applicationSchema = {
...
+            example: 'red',
         },
         icon: {
+            description: `An URL to an icon file to be used for the applications's entry in the application list`,
             type: 'string',
+            example: 'https://github.com/favicon.ico',
         },
         announced: {
+            description:
+                'Internal to Unleash to allow Unleash to know if the application has already been inserted in a previous batch.',
```
Comment (thomasheartman) [inline src/lib/openapi/spec/application-schema.ts:52]: What is a batch here? And what does inserting into one mean?

```suggestion
                'Whether the application has been inserted into a previous batch. Used internally by Unleash.',
```

File: src/lib/openapi/spec/bulk-metrics-schema.ts [hunk 4]
```diff
@@ -11,12 +11,15 @@ export const bulkMetricsSchema = {
         'A batch of metrics accumulated by Edge (or other compatible applications). Includes both application registrations as well usage metrics from clients',
     properties: {
         applications: {
+            description: 'a list of applications registered by Unleash SDK',
```
Comment (thomasheartman) [inline src/lib/openapi/spec/bulk-metrics-schema.ts:14]: I think we're missing an article in front of "unleash sdk". Is it "an"? "the"? It's hard to tell. Or is it plural? 
```suggestion
            description: 'A list of applications registered by (an/the)? Unleash SDK(s?)',
```

File: src/lib/openapi/spec/bulk-metrics-schema.ts [hunk 5]
```diff
@@ -11,12 +11,15 @@ export const bulkMetricsSchema = {
...
             type: 'array',
             items: {
                 $ref: '#/components/schemas/bulkRegistrationSchema',
             },
         },
         metrics: {
+            description:
+                'a list of client usage metrics registered by downstream providers. (Typically Unleash Edge)',
```
Comment (thomasheartman) [inline src/lib/openapi/spec/bulk-metrics-schema.ts:22]:
```suggestion
                'a list of client usage metrics registered by downstream providers (typically Unleash Edge).',
```

File: src/lib/openapi/spec/client-metrics-schema.ts [hunk 6]
```diff
@@ -5,18 +5,48 @@ export const clientMetricsSchema = {
...
+                stop: {
+                    $ref: '#/components/schemas/dateSchema',
+                    description:
+                        'The end of the time window these metrics are valid for. The window is 1 hour wide',
+                    example: '1926-05-08T13:00:00.000Z',
+                },
                 toggles: {
                     type: 'object',
+                    description:
+                        'an object containing feature names with yes/no plus variant usage',
```
Comment (thomasheartman) [inline src/lib/openapi/spec/client-metrics-schema.ts:49]: This sentence contains a lot of words. I'd suggest either breaking it into several sentences or adding more qualifiers:

```suggestion
                        'An object whose keys are feature names and whose values are contain statistics about their evaluation.',
```

File: src/lib/openapi/spec/create-application-schema.ts [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+            example: ['standard', 'gradualRollout', 'mySpecialCustomStrategy'],
+        },
+        url: {
+            description:
+                'A link to reference the application reporting the metrics. Could for instance be a GitHub link to the repository of the application',
+            type: 'string',
+            example: 'https://github.com/Unleash/unleash-client-proxy-js',
+        },
+        color: {
+            description: `Css color to be used to color the application's entry in the application list`,
```
Comment (thomasheartman) [inline src/lib/openapi/spec/create-application-schema.ts:37]:
```suggestion
            description: `The CSS color to be used to color the application's entry in the application list`,
```

File: src/lib/openapi/spec/feature-environment-metrics-schema.ts [hunk 8]
```diff
@@ -1,32 +1,51 @@
 import { FromSchema } from 'json-schema-to-ts';
+import { dateSchema } from './date-schema';
 
 export const featureEnvironmentMetricsSchema = {
     $id: '#/components/schemas/featureEnvironmentMetricsSchema',
     type: 'object',
     additionalProperties: false,
     required: ['environment', 'timestamp', 'yes', 'no'],
+    description:
+        'How many times `feautreName` was evaluated to `true` (yes) and `false` (no) for `appName` in `environmnet`',
```
Comment (thomasheartman) [inline src/lib/openapi/spec/feature-environment-metrics-schema.ts:10]:
```suggestion
        'How many times `featureName` was evaluated to `true` (yes) and `false` (no) for `appName` in `environment`',
```

File: src/lib/routes/admin-api/client-metrics.ts [hunk 9]
```diff
@@ -58,8 +59,11 @@ class ClientMetricsController extends Controller {
                 openApiService.validPath({
                     operationId: 'getRawFeatureMetrics',
                     tags: ['Metrics'],
+                    summary:
+                        'Feature usage metrics for the last 48 hours, grouped by hour',
```
Comment (thomasheartman) [inline src/lib/routes/admin-api/client-metrics.ts:63]: This feels like a mix of a summary and a description, but it's too long for the former and too short for the latter. Can we split it out?

```suggestion
                    summary:
                        'Get feature usage metrics for the last 48 hours',
                    description:
                        'Returns usage metrics from the last 48 hours for the specified feature. The metrics are grouped by hour.',
```

File: src/lib/routes/admin-api/client-metrics.ts [hunk 10]
```diff
@@ -74,8 +78,12 @@ class ClientMetricsController extends Controller {
                 openApiService.validPath({
                     operationId: 'getFeatureUsageSummary',
                     tags: ['Metrics'],
+                    summary: `Last hour of usage and a list of applications that have reported seeing this feature toggle`,
+                    description:
+                        'Separate counts for yes (enabled), no (disabled), as well as how many times each variant was selected during the last hour',
```
Comment (thomasheartman) [inline src/lib/routes/admin-api/client-metrics.ts:83]: Can we make the summary shorter? Much of that information can (and should) be moved into the description.

Is this an accurate representation?

```suggestion
                    summary: 'Get recent metrics for a feature'
                    description: 'Returns evaluation data and a list of applications that have reported seeing this feature toggle within the last hour. The response contains separate counts for yes (enabled), no (disabled), as well as how many times each variant was selected.'
```

File: src/lib/routes/admin-api/metrics.ts [hunk 11]
```diff
@@ -43,10 +47,15 @@ class MetricsController extends Controller {
                 openApiService.validPath({
                     tags: ['Metrics'],
                     operationId: 'createApplication',
+                    summary:
+                        'Create an application to connect reported metrics',
+                    description:
+                        'Is used to report usage as well which sdk the application uses',
```
Comment (thomasheartman) [inline src/lib/routes/admin-api/metrics.ts:53]: What does the description describe? Is that about the created application? And what is this application? I don't know enough about it. Is it documented anywhere?

```suggestion
                    summary:
                        'Create an application to collect metrics',
                    description:
                        'Creates an application in Unleash that is used to bundle metrics. The data collected includes usage data and what SDKs are used.',
```

File: src/lib/routes/admin-api/metrics.ts [hunk 12]
```diff
@@ -60,8 +69,10 @@ class MetricsController extends Controller {
                 openApiService.validPath({
                     tags: ['Metrics'],
                     operationId: 'deleteApplication',
+                    summary: 'Deletes an application',
```
Comment (thomasheartman) [inline src/lib/routes/admin-api/metrics.ts:72]:
```suggestion
                    summary: 'Delete an application',
                    description: 'Delete the application specified in the request URL. Returns 200 OK if the application was successfully deleted or if it didn't exist.'
```

File: src/lib/routes/admin-api/metrics.ts [hunk 13]
```diff
@@ -74,6 +85,9 @@ class MetricsController extends Controller {
             middleware: [
                 openApiService.validPath({
                     tags: ['Metrics'],
+                    summary: 'Gets a list of applications',
+                    description:
+                        'Gets all applications created via metrics reporting and manual creation',
```
Comment (thomasheartman) [inline src/lib/routes/admin-api/metrics.ts:90]: or are there other kinds of applications?

```suggestion
                        'Returns all applications registered with Unleash. Applications can be created via metrics reporting or manual creation',
```

File: src/lib/routes/admin-api/metrics.ts [hunk 14]
```diff
@@ -90,8 +104,12 @@ class MetricsController extends Controller {
                 openApiService.validPath({
                     tags: ['Metrics'],
                     operationId: 'getApplication',
+                    summary: 'Get specific application',
```
Comment (thomasheartman) [inline src/lib/routes/admin-api/metrics.ts:107]:
```suggestion
                    summary: 'Get application data',
```

File: src/lib/routes/admin-api/metrics.ts [hunk 15]
```diff
@@ -90,8 +104,12 @@ class MetricsController extends Controller {
...
+                    description:
+                        'Gets specific application with `appName`. An application contains information on name, sdkVersion and how to display it in a list.',
```
Comment (thomasheartman) [inline src/lib/routes/admin-api/metrics.ts:109]: What is `sdkVersion` here?

```suggestion
                        'Returns data about the specified application (`appName`). The data contains information on the name of the application, sdkVersion, as well as data about how to display this application in a list.',
```

**Review by thomasheartman — COMMENTED** [review]
Nice one! This is very good! 🙌🏼  I have a few comments, but they're mostly questions and suggestions. However, some of them point out missing bits. And regarding missing bits: I think I've seen you leave out one or the other a few times, but both summaries and descriptions are **mandatory** on endpoints/operations.

good going!

File: src/lib/openapi/spec/application-schema.ts (hunk 3 above)
Comment (chriswk) [reply src/lib/openapi/spec/application-schema.ts:52]: Strictly speaking I think we shouldn't expose this, let me clean it up again.

File: src/lib/openapi/spec/application-schema.ts (hunk 2 above)
Comment (chriswk) [reply src/lib/openapi/spec/application-schema.ts:7]: No, they're all optional in the type we use to resolve it as well, except for the appName.

**Review by sighphyre — APPROVED** [review]
LGTM!

**Pull request merged**